near/omni-types/src/lib.rs

```diff
@@ -175,13 +181,43 @@ impl OmniAddress {
             OmniAddress::Base(_) => ChainKind::Base,
         }
     }
+
+    pub fn encode(&self, separator: char, skip_zero_address: bool) -> String {
+        let (chain_str, address) = match self {
+            OmniAddress::Eth(address) => ("eth", address.to_string()),
+            OmniAddress::Near(address) => ("near", address.to_string()),
+            OmniAddress::Sol(address) => ("sol", address.to_string()),
+            OmniAddress::Arb(address) => ("arb", address.to_string()),
+            OmniAddress::Base(address) => ("base", address.to_string()),
+        };
+
+        if skip_zero_address && self.is_zero() {
+            chain_str.to_string()
+        } else {
+            format!("{chain_str}{separator}{address}")
+        }
+    }
+
+    pub fn is_zero(&self) -> bool {
+        match self {
+            OmniAddress::Eth(address) | OmniAddress::Arb(address) | OmniAddress::Base(address) => {
+                address.is_zero()
+            }
+            OmniAddress::Near(address) => address.len() == 0,
+            OmniAddress::Sol(address) => address.is_zero(),
```

Comment:
Just making note that since we're using Wrapped SOL we wouldn't have a zero address

Reply:
Deploy native token has been added here 
ac3979ae5e4efdb1959b13a16fa30739740b885d